### PR TITLE
Respect exception rules in .dockerignore file while build images

### PR DESCRIPTION
**Pull request by marcuslinke**
Respect exception rules like "!Dockerfile" in .dockerignore file while building images. See https://docs.docker.com/reference/builder/#dockerignore-file for details.

### CODE REVIEW

File: src/test/resources/testDockerfileNotIgnored/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,9 @@
+FROM      ubuntu:latest
+
+# Copy testrun.sh files into the container
+
+RUN echo "Successful executed testrun.sh" >	/tmp/testrun.sh
```
Comment (KostyaSha) [inline src/test/resources/testDockerfileNotIgnored/Dockerfile]: echo echo?

Comment (marcuslinke) [inline src/test/resources/testDockerfileNotIgnored/Dockerfile]: Good catch. Thanks!

**Comment by KostyaSha** [comment]
Don't know details, but change LGTM 
Squash && :sheep: it?

**Comment by marcuslinke** [comment]
I tried to squash it but did something wrong. Changes are in master now but contained in a new commit, so I will close this PR. OK?

**Comment by KostyaSha** [comment]
I usually doing 'rebase -i HEAD~5' and then mark as `f` (fixup) all commits that i want squash with previous, then force push

**Comment by KostyaSha** [comment]
I don't see commits in master, just merge this PR?

**Comment by marcuslinke** [comment]
Hmmm. Merge contains no changes. However, next time i will be more carefully. Commit is here: https://github.com/docker-java/docker-java/commit/33fdd7125f3be5aef33f12ca94df9ca28271d120

**Comment by KostyaSha** [comment]
It is, GH sort history by date. And if there is no changes in PR (i.e. merged in other way) GH automatically detect it and marks as merged/closed.

**Pull request merged**